tests/test_fieldset_sampling.py

```diff
@@ -348,3 +348,27 @@ def test_sampling_out_of_bounds_time(mode, allow_time_extrapolation, k_sample_p,
     else:
         with pytest.raises(RuntimeError):
             pset.execute(k_sample_p, starttime=2.0, endtime=2.1, dt=0.1)
+
+
+@pytest.mark.parametrize('mode', ['jit', 'scipy'])
+def test_sampling_multiple_grid_sizes(mode):
```

Comment:
Do we need more unit-tests? E.g. that adding a new `Field` with a `Grid`that already exists means that no new `Grid` is added (i.e. test the `existing_grid` functionality in `GridSet`)?